src/rockstor/system/pkg_mgmt.py

```diff
@@ -33,78 +34,81 @@
 
 logger = logging.getLogger(__name__)
 
-YUM = '/usr/bin/yum'
-RPM = '/usr/bin/rpm'
-SYSTEMCTL = '/usr/bin/systemctl'
-AT = '/usr/bin/at'
-YCFILE = '/etc/yum/yum-cron.conf'  # Doesn't exist in openSUSE
+YUM = "/usr/bin/yum"
+RPM = "/usr/bin/rpm"
+ZYPPER = "/usr/bin/zypper"
+SYSTEMCTL = "/usr/bin/systemctl"
+AT = "/usr/bin/at"
+YCFILE = "/etc/yum/yum-cron.conf"  # Doesn't exist in openSUSE
 
 
 def install_pkg(name):
-    return run_command([YUM, '--setopt=timeout=600', '-y', 'install', name])
+    return run_command([YUM, "--setopt=timeout=600", "-y", "install", name])
 
 
 def downgrade_pkgs(*packages):
-    cmd = [YUM, '--setopt=timeout=600', '-y', 'downgrade', ]
+    cmd = [YUM, "--setopt=timeout=600", "-y", "downgrade"]
     for p in packages:
         cmd.append(p)
     return run_command(cmd)
 
 
 def auto_update(enable=True):
-    service = 'yum-cron'
+    # TODO: Add openSUSE zypper equivalent
+    service = "yum-cron"
     fo, npath = mkstemp()
     updated = False
-    with open(YCFILE) as ifo, open(npath, 'w') as tfo:
+    with open(YCFILE) as ifo, open(npath, "w") as tfo:
         for line in ifo.readlines():
-            if (re.match('apply_updates = ', line) is not None):
-                if (enable):
-                    tfo.write('apply_updates = yes\n')
+            if re.match("apply_updates = ", line) is not None:
+                if enable:
+                    tfo.write("apply_updates = yes\n")
                 else:
-                    tfo.write('apply_updates = no\n')
+                    tfo.write("apply_updates = no\n")
                 updated = True
             else:
                 tfo.write(line)
-    if (not updated):
-        raise Exception('apply_updates directive missing in %s, assuming its '
-                        'is corrupt. No change made.' % YCFILE)
+    if not updated:
+        raise Exception(
+            "apply_updates directive missing in {}, assuming its "
```

Comment:
Minor typo:
```suggestion
            "apply_updates directive missing in {}, assuming it "
```

Reply:
@FroggyFlox Thanks for this.
Not one of mine surprisingly but good to have it in anyway. Committing suggestion. Tricky with us doing simultaneous Black formatting. We will have to keep that to small files only when rolled into other pull requests.